src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -9066,17 +9066,11 @@ instruct minI_reg_reg(iRegINoSp dst, iRegI src)
   match(Set dst (MinI dst src));
 
   ins_cost(BRANCH_COST + ALU_COST);
-  format %{
-    "ble $dst, $src, skip\t#@minI_reg_reg\n\t"
-    "mv  $dst, $src\n\t"
-    "skip:"
-  %}
+  format %{"minI_reg_reg $dst, $dst, $src\t#@minI_reg_reg\n\t"%}
 
   ins_encode %{
-    Label Lskip;
-    __ ble(as_Register($dst$$reg), as_Register($src$$reg), Lskip);
-    __ mv(as_Register($dst$$reg), as_Register($src$$reg));
-    __ bind(Lskip);
+    __ cmov_gt(as_Register($dst$$reg), as_Register($src$$reg),
+               as_Register($dst$$reg), as_Register($src$$reg));
```

Comment:
But the `ins_cost` isn't updated to reflect this change? It is still `BRANCH_COST + ALU_COST` which will only reflect the branch code. Seems better to create seperate match rules for these cmove cases with `UseZicond` as the predicate and proper costs. We already have two sets of match rules for Max and Min. The other one which should be the most efficient is in file `riscv_b.ad` [1]. Do we have a hardware which implements `Zicond` but doesn't have `Zbb`?

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/riscv_b.ad#L436

Reply:
Yes, better to have seperate match rules.

But I also agree that it's a bit messy and unclear how many will actually use that version.
And all of these versions increase test cost.
I think we should aim for RVA23 as the well supported and tested set of extensions.

Reply:
For performance consideration please check existing `CMoveI`, which calls `enc_cmove` which calls `cmov_xx`. So if `CMoveI` brings benefit when `UseZicond == true` over `UseZicond != true`, this refactoring should also works expected, as they use the same code.

As for the question, whether a hardware will support `zicond` but not `zbb`, I have no answer.

But, anyway in fact you can just consider this as a code cleanup, in this sense seems it should be good?

Reply:
> better to have seperate match rules.

We could also do the similar thing to `CMoveX`, for this part I can do it in a separate PR together if this one is accecpted.

Reply:
Yes, good point, I think we need to update the cost for some others as well.
We can do that separately.

Reply:
I'm pretty sure the p550(hifive primier p550) doesn't support zicond but does support zbb, zba

Reply:
> But, anyway in fact you can just consider this as a code cleanup, in this sense seems it should be good?

Seems fine to me in that respect. We can fix the cost in another PR.